File: src/main/java/com/lannstark/lec12/java/Lec12Main.java
package com.lannstark.lec12.java;

import com.lannstark.lec12.Person;

public class Lec12Main {

    public static void main(String[] args) {
        // Person.Companion.newBaby("ABC");  // 이름이 없을때
        // Person.newBaby("ABC");            // @JvmStatic 사용시
        // Person.Factory.newBaby("ABC");    // 이름 존재 시

        moveSomething(new Movable() {

            @Override
            public void move() {
                System.out.println("움직인다~~");
            }

            @Override
            public void fly() {
                System.out.println("난다~~");
            }
        });
    }

    private static void moveSomething(Movable movable) {
        movable.move();
        movable.fly();
    }

}

File: src/main/java/com/javaToKotlin/lec12/Lec12Main.kt
package com.javaToKotlin.lec12

import com.javaToKotlin.lec12.java.Movable

fun main() {
    moveSomething(object : Movable {
        override fun move() {
            println("움직인다~~")
        }

        override fun fly() {
            println("난다~~")
        }

    })
}

private fun moveSomething(movable: Movable) {
    movable.move()
    movable.fly()
}